### DISCUSSION TITLE
managing background

### DISCUSSION THREAD

**Discussion by ilune** [hako-mikan/sd-webui-regional-prompter#124] [Q&A] [answered]
I want to generate the image of a woman with a background of a battle, something that should be totally different

```
logo of a woman with a background BREAK
woman caucasian redhead, medieval, dress ,woman BREAK
 medieval battle with sword and blood pencil style gray gradients ,background
```

I selected the divide mode «prompt», but it seems not to work, i have the woman with a sword and blood on her ...
With prompt ex, it doesn't work at all.

Any ideas how I can do this ?

**Comment by Symbiomatrix** [comment]
Are you getting 2 additional outputs?

**Reply by ilune** [reply]
> Are you getting 2 additional outputs?

Yes, quite similar, you could try.

**Reply by Symbiomatrix** [reply]
I have. Not seeing any technical issue, with a threshold of 0.6. The woman and background are properly separated. In meinapastel, there seems to be no concept of "battle" / "pencil style" or anything like that, so it just makes grey background or a bloody sword. Redhead, medieval dress, caucasian, pretty accurate. Logo, absolutely.

In promptex mode, since you picked a very general keyword like "background" which encompasses the whole image, I think it overrides the "logo" and adds grey on the outside.

If you can't get it in regular prompting, it won't work in regions.

![RegionalPmode](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/41131377/e5af2f92-0b4d-480a-a5d6-7ca4d7d04dd9)

![RegionalPmodeB](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/41131377/452d35c3-13d9-4ab5-845a-cab05cb6c660)

**Reply by ilune** [reply]
For my part, the problem is that he make a warrior woman fighter, i want to separate the two concepts. I don't want a woman with a sword, i want a woman with a dress, and a battle behind ^^.

Perhaps i misunderstand how this script works, i don't know how to do this, by the way, finding this script change lot of things in image generation, because stable diffusion tends to mix all keywords without logic...

**Reply by Symbiomatrix** [reply]
You're not following what I'm saying.
What result are you getting for just ` medieval battle with sword and blood` ?

**Reply by ilune** [reply]
Hum, 

This 

```
medieval battle with sword and blood pencil style gray gradients 
Negative prompt: easynegative
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 1239471019, Size: 512x512, Model hash: e04b020012, Model: rpg_V4, Version: v1.2.1
``` 

Gives this

![00147-1239471019](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/13850216/361a5e21-8487-4072-8b04-7142161f47d5)

If i change text a little

```
medieval battle with sword and blood, luis royo, pencil art
Negative prompt: easynegative
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 1754467351, Size: 512x512, Model hash: e04b020012, Model: rpg_V4, Version: v1.2.1
```
It's better

![00142-1754467351](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/13850216/2ef1b127-c801-42e4-aa59-3719be66ab62)

Without selecting region prompt this 

```
logo of a woman with a background BREAK
woman caucasian redhead, medieval, dress ,woman BREAK 
medieval battle with sword and blood, luis royo, pencil art ,background

Negative prompt: easynegative
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 1777564987, Size: 512x512, Model hash: e04b020012, Model: rpg_V4, Version: v1.2.1
```

gives this 

![00143-1777564987](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/13850216/12238f29-c441-423b-ba19-93ee1c78bacf)

The same with region prompt


```
logo of a woman with a background BREAK
woman caucasian redhead, medieval, dress ,woman BREAK 
medieval battle with sword and blood, luis royo, pencil art ,background
Negative prompt: easynegative
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 725000859, Size: 512x512, Model hash: e04b020012, Model: rpg_V4, Version: v1.2.1, RP Active: True, RP Divide mode: Prompt, RP Calc Mode: Attention, RP Ratios: "1,3; 1,1,1,1; 2,1,1,1; 1,1,1,1;1,3", RP Base Ratios: 0.3, RP Use Base: True, RP Use Common: False, RP Use Ncommon: False, RP Change AND: False, RP LoRA Neg Te Ratios: 0, RP LoRA Neg U Ratios: 0, RP threshold: 0.4
```
gives

![00150-725000859](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/13850216/f48c570f-cb2e-480d-b15b-30ac943852f0)

**Reply by Symbiomatrix** [reply]
I think some concept bleed is inevitable. Unlike multi subject render and others, prompt mode seems to train the subprompt alongside the main prompt at every step, so if your scene is a battle you get a warrior woman because that's what the model knows should happen in a battle, I believe.
I'm not entirely sure whether the negs are connected respectively, might be useful to remove violence from the woman subprompt.
"Logo of" is clearly overpowering the scene's style, without it the results are at least passably close.

Adding distance and a non warrior woman (noble / peasant) helps a little.
![RegionalPmodeC](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/41131377/ab301f1c-dceb-49d7-abbb-10f268f5f6f9)

**Reply by ilune** [reply, marked as answer]
Thank you for you time, I search a little more and get something using this:

```
close-up view of Tanaelle, fields in the distance in a a circle frame BREAK
woman caucasian redhead, medieval, dress ,Tanaelle BREAK 
medieval battle of armies in the fields, dark clouds, lightning, luis royo, pencil art ,fields 
```

I don't use «woman» but a name, i don't speak of battle but of fields. Not perfect but i have some success ^^.

**Reply by Symbiomatrix** [reply]
Aye, that's pretty much the technique, it seems. Personally I still prefer my regions to be predictable, but it's lightyears ahead of cutoff.
